File: QueueManager/src/client/java/es/gabirp/queuemanager/websocket/WebSocketMessage.java
package es.gabirp.queuemanager.websocket;

import org.jetbrains.annotations.Nullable;

public class WebSocketMessage {
    public MessageType type;

    @Nullable
    public String content;

    public int queuePosition;

    public WebSocketMessage(MessageType type, int queuePosition, @Nullable String content)
    {
        this.type = type;
        this.content = content;
        this.queuePosition = queuePosition;
    }
}

File: QueueManager.Discord/QueueManager.DiscordBot/Websocket/WebsocketMessage.cs
namespace QueueManager.DiscordBot.Websocket;

public class WebsocketMessage
{
    public MessageType type { get; set; }

    public string? content { get; set; }

    public int queuePosition { get; set; }

    public WebsocketMessage(MessageType type, int queuePosition, string content)
    {
        this.type = type;
        this.content = content;
        this.queuePosition = queuePosition;
    }
}

public enum MessageType
{
    HEARTBEAT = 0,
    HEARTBEAT_RESPONSE = 1,
    QUEUEUPDATE = 2,
    ENDQUEUE = 3,
    ERROR = 4,
    BYE = 5,
    SERVERDISCONNECT = 6,
}